lang/java/avro/src/main/java/org/apache/avro/file/DataFileWriter.java

```diff
@@ -195,25 +195,31 @@ public boolean isFlushOnEveryBlock() {
   }
 
   /** Open a writer appending to an existing file.
+   * <strong>This method closes <code>in</code>.</strong>
    * @param in reading the existing file.
    * @param out positioned at the end of the existing file.
    */
   public DataFileWriter<D> appendTo(SeekableInput in, OutputStream out)
     throws IOException {
     assertNotOpen();
-    DataFileReader<D> reader =
-      new DataFileReader<D>(in, new GenericDatumReader<D>());
-    this.schema = reader.getSchema();
-    this.sync = reader.getHeader().sync;
-    this.meta.putAll(reader.getHeader().meta);
-    byte[] codecBytes = this.meta.get(DataFileConstants.CODEC);
-    if (codecBytes != null) {
-      String strCodec = new String(codecBytes, "UTF-8");
-      this.codec = CodecFactory.fromString(strCodec).createInstance();
-    } else {
-      this.codec = CodecFactory.nullCodec().createInstance();
+    DataFileReader<D> reader = null;
+    try {
+      reader = new DataFileReader<D>(in, new GenericDatumReader<D>());
+      this.schema = reader.getSchema();
+      this.sync = reader.getHeader().sync;
+      this.meta.putAll(reader.getHeader().meta);
+      byte[] codecBytes = this.meta.get(DataFileConstants.CODEC);
+      if (codecBytes != null) {
+        String strCodec = new String(codecBytes, "UTF-8");
+        this.codec = CodecFactory.fromString(strCodec).createInstance();
+      } else {
+        this.codec = CodecFactory.nullCodec().createInstance();
+      }
+    } finally {
+      if (reader != null)
+        reader.close();
+      in.close();
```

Comment:
You're right, it should be closed by the caller but it is already implemented this way as we are closing the reader which closes the SeekableInput as well. I did not want to break backward compatibility but fix the actual situation that the resources are not closed every time.
That's why I've tried to highlight in the method comment that this method closes the specified SeekableInput.